webapp/components/features/FollowList/FollowList.spec.js

```diff
@@ -0,0 +1,150 @@
+import { config, mount } from '@vue/test-utils'
+import Vuex from 'vuex'
+
+import helpers from '~/storybook/helpers'
+import FollowList from './FollowList.vue'
+
+const localVue = global.localVue
+
+config.stubs['client-only'] = '<span><slot /></span>'
+config.stubs['ds-space'] = '<span><slot /></span>'
+config.stubs['nuxt-link'] = '<span><slot /></span>'
+
+const user = {
+  ...helpers.fakeUser()[0],
+  followedByCount: 12,
+  followingCount: 15,
+  followedBy: helpers.fakeUser(7),
+  following: helpers.fakeUser(7),
+}
+
+const allConnectionsUser = {
+  ...user,
+  followedBy: [
+    ...user.followedBy,
+    ...helpers.fakeUser(user.followedByCount - user.followedBy.length),
+  ],
+  following: [...user.following, ...helpers.fakeUser(user.followingCount - user.following.length)],
+}
+
+const noConnectionsUser = {
+  ...user,
+  followedByCount: 0,
+  followingCount: 0,
+  followedBy: [],
+  following: [],
+}
+
+describe('FollowList.vue', () => {
+  let store, getters
+  const Wrapper = (customProps) =>
+    mount(FollowList, {
+      store,
+      propsData: { user, ...customProps },
+      mocks: {
+        $t: jest.fn((str) => str),
+      },
+      localVue,
+    })
+
+  beforeAll(() => {
+    getters = {
+      'auth/user': () => {
+        return {}
+      },
+      'auth/isModerator': () => false,
+    }
+  })
+
+  describe('mount', () => {
+    beforeAll(() => {
+      store = new Vuex.Store({
+        getters,
+      })
+    })
+
+    describe('given a type', () => {
+      describe('of `following`', () => {
+        it('uses the `following` data on :user', () => {
+          const wrapper = Wrapper({ user, type: 'following' })
+          wrapper.find('.base-button').trigger('click')
+
+          expect(wrapper.vm.allConnectionsCount).toBe(user.followingCount)
+          expect(wrapper.findAll('.user-teaser')).toHaveLength(user.following.length)
+          expect(wrapper.emitted('fetchAllConnections')).toEqual([['following']])
+        })
+      })
+
+      describe('of `followedBy`', () => {
+        it('uses the `followedBy` data on :user', () => {
+          const wrapper = Wrapper({ type: 'followedBy' })
+          wrapper.find('.base-button').trigger('click')
+
+          expect(wrapper.vm.allConnectionsCount).toBe(user.followedByCount)
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2020-03-26T11:52:20Z" title="Thursday, March 26th 2020, 12:52:20 pm +01:00">Mar 26, 2020</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -0,0 +1,129 @@
+import { config, mount } from '@vue/test-utils'
+import Vuex from 'vuex'
+
+import helpers from '~/storybook/helpers'
+import FollowList from './FollowList.vue'
+
+const localVue = global.localVue
+
+config.stubs['client-only'] = '<span><slot /></span>'
+config.stubs['ds-space'] = '<span><slot /></span>'
+config.stubs['nuxt-link'] = '<span><slot /></span>'
+
+describe('FollowList.vue', () => {
+  let store, mocks, getters, propsData
+
+  beforeAll(() => {
+    mocks = {
+      $t: jest.fn(),
+    }
+    getters = {
+      'auth/user': () => {
+        return {}
+      },
+      'auth/isModerator': () => false,
+    }
+    const [_user] = helpers.fakeUser()
+    propsData = {
+      user: {
+        ..._user,
+        followedByCount: 12,
+        followingCount: 15,
+        followedBy: helpers.fakeUser(7),
+        following: helpers.fakeUser(7),
+      },
+    }
+  })
+
+  describe('mount', () => {
+    beforeAll(() => {
+      store = new Vuex.Store({
+        getters,
+      })
+    })
+
+    describe('given a user with connections', () => {
+      ;['following', 'followedBy'].forEach((type) =>
+        describe(`and type=${type}`, () => {
+          let wrapper
+          let queryMock
+
+          beforeAll(() => {
+            queryMock = jest.fn().mockResolvedValue({
+              data: { User: [{ [type]: additionalConnections[type] }] },
+            })
+
+            wrapper = mount(FollowList, {
+              store,
+              propsData: { ...propsData, type: type },
+              mocks: {
+                ...mocks,
+                $apollo: {
+                  query: queryMock,
+                },
+              },
+              localVue,
+            })
+          })
+
+          it(`shows the users ${type}`, () => {
+            expect(wrapper.findAll('.user-teaser').length).toEqual(propsData.user[type].length)
+          })
+
+          it(`has a button to load all remaining users ${type}`, async () => {
+            jest.useFakeTimers()
+
+            wrapper.find('button').trigger('click')
+            await jest.runAllTicks()
+            await wrapper.vm.$nextTick()
+
+            expect(wrapper.vm.connections.length).toBe(propsData.user[`${type}Count`])
+            expect(queryMock).toHaveBeenCalledWith({
+              query: wrapper.vm.queries[type],
+              variables: { id: propsData.user.id },
```


I kind of like this, but I've gotten feedback from @roschaefer that it's safer you just use hard-coded values here.

Reply:
> [<img alt="rbeer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/rbeer) **Authored by [rbeer](https://github.com/rbeer)**
_<time datetime="2020-03-30T06:23:29Z" title="Monday, March 30th 2020, 8:23:29 am +02:00">Mar 30, 2020</time>_

---


removed, because not needed anymore in [e6361c9](https://github.com/Human-Connection/Human-Connection/pull/3293/commits/e6361c9b740f104c0a0d86b463be835f10311f3d#diff-970486bf70217f927b18aadea0c8b1b4)

In this case would have been necessary, since the user is generated by `faker`, i.e. the id not known before the test runs.